src/compiler/factory.ts

```diff
@@ -2292,15 +2292,26 @@ namespace ts {
     /**
      * Sets a custom text range to use when emitting source maps.
      */
-    export function setSourceMapRange<T extends Node>(node: T, range: TextRange | undefined) {
+    export function setSourceMapRange<T extends Node>(node: T, range: SourceMapRange | undefined) {
         getOrCreateEmitNode(node).sourceMapRange = range;
         return node;
     }
 
     /**
+     * Create an external source map source file reference
+     */
+    export function createSourceMapSource(fileName: string, text: string, skipTrivia?: (pos: number) => number) {

```

Comment:
To make this clean I believe there are three options:
1) Remove the extension of `SourceFileLike` in `service/types.ts` and fix all calls to the method `getLineAndCharacterofPosition()` to call the function `getLineAndCharacterOfPosition()`.
2) Introduce a new sub-type of `SourceFileLIke` in `service/types.ts`, such as `ServiceSource`, that extends `SourceFileLIke` and move all references to `SourceFileLIke` in `services` to use `ServiceSource`.
3) Introduce a new type in `compiler/types.ts` called something like `SourceWithLines`  that  `SourceFileLike` extends from and change all other references to `SourceFileLike` in `compiler` to  `SourceWithLines`.

I like the first one best. It is faster than using a method and I could not find any cases in `service` that overrides `getCharacterAndLineOfPosition` in a source file that would required either (2) or (3).